src/components/organisms/forms/InstitutionForm/InstitutionForm.tsx

```diff
@@ -0,0 +1,189 @@
+import { FC, useCallback, useMemo, useState } from 'react'
+import { useForm, SubmitHandler, FieldPath } from 'react-hook-form'
+import DynamicForm, {
+  FieldProps,
+  InputTypes,
+} from 'components/organisms/DynamicForm/DynamicForm'
+import { useTranslation } from 'react-i18next'
+import FormButtons from 'components/organisms/FormButtons/FormButtons'
+import { includes, join, map } from 'lodash'
+import { Privileges } from 'types/privileges'
+import classes from './classes.module.scss'
+import useAuth from 'hooks/useAuth'
+import useValidators from 'hooks/useValidators'
+import { showNotification } from 'components/organisms/NotificationRoot/NotificationRoot'
+import { NotificationTypes } from 'components/molecules/Notification/Notification'
+import { ValidationError } from 'api/errorHandler'
+import { useUpdateInstitution } from 'hooks/requests/useInstitutions'
+import { InstitutionPostType, InstitutionType } from 'types/institutions'
+import Button from 'components/molecules/Button/Button'
+import classNames from 'classnames'
+
+interface FormValues {
+  name: string
+  short_name?: string | null
+  email?: string | null
+  phone?: string | null
+}
+
+type InstitutionFormProps = InstitutionType
+
+const InstitutionForm: FC<InstitutionFormProps> = ({
+  id,
+  name,
+  short_name,
+  email,
+  phone,
+}) => {
+  // hooks
+  const { t } = useTranslation()
+  const [isUpdatingData, setIsUpdatingData] = useState(false)
+  const { userPrivileges } = useAuth()
+  const { emailValidator, phoneValidator } = useValidators()
+  const { updateInstitution, isLoading } = useUpdateInstitution({
+    institutionId: id,
+  })
+
+  const defaultValues = useMemo(
+    () => ({
+      name,
+      short_name,
+      email,
+      phone,
+    }),
+    [email, name, phone, short_name]
+  )
+  const {
+    control,
+    handleSubmit,
+    reset,
+    formState: { isSubmitting, isValid },
+    setError,
+  } = useForm<FormValues>({
+    reValidateMode: 'onSubmit',
+    values: defaultValues,
+  })
+
+  const fields: FieldProps<FormValues>[] = [
+    {
+      inputType: InputTypes.Text,
+      ariaLabel: t('label.institution'),
+      label: t('label.institution'),
+      disabled: !isUpdatingData,
+      name: 'name',
+      className: classNames(classes.inputInternalPosition, {
```

Comment:
**[non-blocker]**

Let's leave this for now, but we might change it later. We have the DisplayValue component which will be displayed instead of the actual input, when we pass "onlyDisplay: true" as a prop here. It was meant for similar usecases, however it doesn't show the label currently, so would need to be modified.

If we will deal with any future places, where we need this again, then it might make sense to do the modification for the DisplayValue component, but not right now